### ISSUE TITLE
Unexplainable error with SpectralResidual: RuntimeWarning: invalid value encountered in subtract   res_amp = log_amp - ma_log_amp

### ISSUE THREAD

**Issue by rajdeep-biswas** [SeldonIO/alibi-detect#291]
This happens in close causation from #289 
Way to reproduce is as follows -

```
X = array([    10,     11,     12, ..., 175197, 175198, 175199])
t = array([2, 7, 1, ..., 2, 2, 2])

od = SpectralResidual(
    threshold=1.,
    window_amp=20,
    window_local=20,
    n_est_points=10,
    n_grad_points=5
)

od.infer_threshold(
    X,
    t=t,  # array with timesteps, assumes dt=1 between observations if omitted
    threshold_perc=95
)
```

And as a results, it causes od.predict() to return `instance_score`s full of nans. As shown in screenshot below -

<img width="1094" alt="Screenshot 2021-07-14 at 1 06 26 AM" src="https://user-images.githubusercontent.com/32306614/125514304-89db52d6-3fdd-43ca-8d21-f47bee8972d2.png">

**Comment by RobertSamoilescu** [comment]
@therajdeepbiswas, please take a look at: https://github.com/SeldonIO/alibi-detect/issues/289#issuecomment-974055382
I believe is the same issue.

**Comment by arnaudvl** [comment]
Addressed by #396
